coordinator/src/db/funding_rates.rs

```diff
@@ -0,0 +1,89 @@
+use crate::schema::funding_rates;
+use anyhow::Context;
+use anyhow::Result;
+use diesel::prelude::*;
+use rust_decimal::prelude::FromPrimitive;
+use rust_decimal::prelude::ToPrimitive;
+use rust_decimal::Decimal;
+use time::OffsetDateTime;
+use xxi_node::commons::to_nearest_hour_in_the_past;
+
+#[derive(Queryable, Debug)]
+struct FundingRate {
+    #[diesel(column_name = "id")]
+    _id: i32,
+    start_date: OffsetDateTime,
+    end_date: OffsetDateTime,
+    rate: f32,
+    #[diesel(column_name = "timestamp")]
+    _timestamp: OffsetDateTime,
+}
+
+pub(crate) fn insert(
+    conn: &mut PgConnection,
+    funding_rates: &[xxi_node::commons::FundingRate],
+) -> Result<()> {
+    conn.transaction(|conn| {
+        for funding_rate in funding_rates {
+            insert_one(conn, funding_rate)
+                .with_context(|| format!("Failed to insert funding rate: {funding_rate:?}"))?;
+        }
+
+        Ok(())
+    })
+}
+
+fn insert_one(conn: &mut PgConnection, params: &xxi_node::commons::FundingRate) -> QueryResult<()> {
+    let affected_rows = diesel::insert_into(funding_rates::table)
+        .values(&(
+            funding_rates::start_date.eq(params.start_date()),
+            funding_rates::end_date.eq(params.end_date()),
+            funding_rates::rate.eq(params.rate().to_f32().expect("to fit")),
+        ))
+        .execute(conn)?;
+
+    if affected_rows == 0 {
+        return Err(diesel::result::Error::NotFound);
+    }
+
+    Ok(())
+}
+
+// TODO: Check that `end_date` isn't already in the past.
```

Comment:
Not outdated, but I'm going to remove it because I think the app should display it anyway. If we show the time, it could say e.g. `0.03% 10 minutes ago`.